gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -608,19 +611,20 @@ private final PreparedStatement prepMinion(
 						if (prepareIsolationLevel == Connection.TRANSACTION_NONE) {
 							cc.markAsDDLForSnappyUse(true);
 							return getPreparedStatementForSnappy(false, statementContext, lcc,
-                  cc.isMarkedAsDDLForSnappyUse(), checkCancellation);
+                  cc.isMarkedAsDDLForSnappyUse(), checkCancellation, false);
 						}
 					}
 					qt = p.parseStatement(getQueryStringForParse(lcc), paramDefaults);
 				}
 				catch (StandardException | AssertFailure ex) {
           //wait till the query hint is examined before throwing exceptions or
-          if (routeQuery && !DML_TABLE_PATTERN.matcher(source).matches()) {
+          if (routeQuery && !INSERT_TABLE_PATTERN.matcher(source).matches()) {
```

Comment:
Update/delete on row tables should not be routed. How are you taking care of that?